contracts/0.8.9/LidoOracleNew.sol

```diff
@@ -0,0 +1,571 @@
+     */
+    function getBeaconReportReceiver() external view returns (address) {
+        return BEACON_REPORT_RECEIVER_POSITION.getStorageAddress();
+    }
+
+    /**
+     * @notice Return the current reporting array element with index `_index`
+     */
+    function getMemberReport(uint256 _index)
+        external
+        view
+        returns (
+            MemberReport memory report
+        )
+    {
+        report = _decodeReport(distinctReports[_index]);
+    }
+
+    /**
+     * @notice Set the receiver contract address to `_address` to be called when the report is pushed
+     * @dev Specify 0 to disable this functionality
+     */
+    function setBeaconReportReceiver(address _address)
+        external onlyRole(SET_BEACON_REPORT_RECEIVER_ROLE)
+    {
+        if(_address != address(0)) {
+            IBeaconReportReceiver iBeacon;
+            if (!_address.supportsInterface(iBeacon.processLidoOracleReport.selector)) {
+                revert BadBeaconReportReceiver();
+            }
+        }
+
+        BEACON_REPORT_RECEIVER_POSITION.setStorageAddress(_address);
+        emit BeaconReportReceiverSet(_address);
+    }
+
+    /**
+     * @notice Return the initialized version of this contract starting from 0
+     */
+    function getVersion() external view returns (uint256) {
+        return CONTRACT_VERSION_POSITION.getStorageUint256();
+    }
+
+    /**
+     * @notice Report beacon balance and its change during the last frame
+     */
+    function getLastCompletedReportDelta()
+        external
+        view
+        returns (
+            uint256 postTotalPooledEther,
+            uint256 preTotalPooledEther,
+            uint256 timeElapsed
+        )
+    {
+        postTotalPooledEther = POST_COMPLETED_TOTAL_POOLED_ETHER_POSITION.getStorageUint256();
+        preTotalPooledEther = PRE_COMPLETED_TOTAL_POOLED_ETHER_POSITION.getStorageUint256();
+        timeElapsed = TIME_ELAPSED_POSITION.getStorageUint256();
+    }
+
+    /**
+     * @notice Return last completed epoch
+     */
+    function getLastCompletedEpochId() external view returns (uint256) {
+        return LAST_COMPLETED_EPOCH_ID_POSITION.getStorageUint256();
+    }
+
+    /**
+     * @notice Add `_member` to the oracle member committee list
+     */
+    function addOracleMember(address _member)
+        external onlyRole(MANAGE_MEMBERS_ROLE)
+    {
+        _addOracleMember(_member);
+    }
+
+    /**
+     * @notice Remove '_member` from the oracle member committee list
+     */
+    function removeOracleMember(address _member)
+        external onlyRole(MANAGE_MEMBERS_ROLE)
+    {
+        _removeOracleMember(_member);
+    }
+
+    function handleCommitteeMemberReport(
+        MemberReport calldata _report
+    ) external {
+        BeaconSpec memory beaconSpec = _getBeaconSpec();
+        bool hasEpochAdvanced = _validateAndUpdateExpectedEpoch(_report.epochId, beaconSpec);
```

Comment:
looks like the same member can advance the epoch preventing for reports finalization
maybe address this by setting cooldown time or checking the epoch number against block number / timestamp?